pkg/utils/selector.go

```diff
@@ -360,8 +366,44 @@ func filterByPhaseSelector(pods []v1.Pod, phases labels.Selector) ([]v1.Pod, err
 	return filteredList, nil
 }
 
-// filterByNamespaces filters a list of pods by a given namespace selector.
-func filterByNamespaces(pods []v1.Pod, namespaces labels.Selector) ([]v1.Pod, error) {
+func filterByNamespaces(pods []v1.Pod) []v1.Pod {
+	var filteredList []v1.Pod
+
+	for _, pod := range pods {
+		if IsAllowedNamespaces(pod.Namespace) {
+			filteredList = append(filteredList, pod)
+		} else {
+			log.Info("filter pod by namespaces",
+				"pod", pod.Name, "namespace", pod.Namespace)
+		}
+	}
+	return filteredList
+}
+
+// IsAllowedNamespaces return whether namespace allows the execution of a chaos task
+func IsAllowedNamespaces(namespace string) bool {
+	if len(AllowedNamespaces) > 0 {
+		for _, ns := range AllowedNamespaces {
+			if strings.HasPrefix(namespace, ns) || strings.HasSuffix(namespace, ns) {
```

Comment:
Sorry, but I don't really understand why do we need `HasPrefix` and `HasSuffix` here? Is `namespace==ns` not enough? 

I think we only want to support equal match.

Reply:
> Sorry, but I don't really understand why do we need `HasPrefix` and `HasSuffix` here? Is `namespace==ns` not enough?
> 
> I think we only want to support equal match.

Sometimes ,We may deploy three environments for Development / Integration / Testing in the same cluster.
We may use label or prefix or suffix to distinguish between different environments.
So I think support equal match is not enough. @yeya24

Reply:
Sorry but I disagree with this. The __fix matching is dangerous and this shouldn't be a default behavior.

Reply:
Perhaps we can provide a configuration item to select an  equal match or a __fix match? @yeya24

Reply:
If you want to support __fix, I would say it is better to support regex directly. __fix is not clear enough.
IMO equal matches can already cover most of the cases. Let's get @cwen0 's opinion.

Reply:
We have about 200 namespaces in a k8s cluster, and they are the Development/Integration/Testing environment for business. If only support equal matches，We need to set a lot of namespace in `AllowedNamespaces`. If we create a new namespace, we also need to update this.

Reply:
How about using simple regular expressions?

Reply:
@cwen0 @yeya24 

OK,I'll re-implement it using regular expressions.